Objects/abstract.c

```diff
@@ -495,6 +495,48 @@ _Py_add_one_to_index_C(int nd, Py_ssize_t *index, const Py_ssize_t *shape)
     }
 }
 
+Py_ssize_t
+PyBuffer_SizeFromFormat(const char *format)
+{
+    PyObject *structmodule = NULL;
+    PyObject *calcsize = NULL;
+    PyObject *res = NULL;
+    PyObject *fmt = NULL;
+    Py_ssize_t itemsize = -1;
+
+    structmodule = PyImport_ImportModule("struct");
+    if (structmodule == NULL) {
+        goto done;
```

Comment:
If this `goto done;` is executed, a segfault will occur at `Py_DECREF(structmodule);`.

Reply:
How did you reproduce the segfault? I suspected a leak but verified none. I just ran one of the tests that use this function. using `./python -m test -R 3:3 test_buffer -m test_pybuffer_size_from_format`

Reply:
You can add "structmodule = NULL;" before the if to reproduce the crash.

Reply:
The segfault would only occur if `PyImport_ImportModule()` failed (which is unlikely, but possible). It is enough to just use `return -1;` here instead of the `goto`.